handler/handler.go

```diff
@@ -0,0 +1,118 @@
+package handler
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/yseto/sabatrapd/charset"
+	"github.com/yseto/sabatrapd/config"
+	"github.com/yseto/sabatrapd/notification"
+	"github.com/yseto/sabatrapd/smi"
+	"github.com/yseto/sabatrapd/template"
+
+	g "github.com/gosnmp/gosnmp"
+	"github.com/sleepinggenius2/gosmi/types"
+)
+
+const SnmpTrapOIDPrefix = ".1.3.6.1.6.3.1.1.4.1"
+
+type Handler struct {
+	Config    *config.Config
+	Queue     *notification.Queue
+	MibParser *smi.SMI
+	Decoder   *charset.Decoder
+}
+
+func (h *Handler) OnNewTrap(packet *g.SnmpPacket, addr *net.UDPAddr) {
+	// log.Printf("got trapdata from %s\n", addr.IP)
+	config := h.Config
+
+	if config.TrapServer.Community != packet.Community {
```

Comment:
```suggestion
	if config.TrapServer.Community != "" && config.TrapServer.Community != packet.Community {
```

コミュニティ明記時のみ照合検査、としてみようかなと。

Reply:
この変更はよさそうです